### ISSUE TITLE
Quiet mode?  (suppress all output when no offenses)

### ISSUE THREAD

**Issue by dhempy** [bbatsov/rubocop#1583]
Is there a way to have rubocop output nothing if no offenses are found?

I'm going to write a formatter to do this, but it seems like this would be a good command line option. 

If I've overlooked this functionality already in rubocop, please let me know.

**Comment by Koronen** [comment]
What's the use case for such an option?

One alternate solution I see is to redirect all output to a file and later read it if necessary. You can check RuboCop's exit code to see if there were any offences.

```sh
rubocop > /tmp/rubocop.log
[ $? -ne 0 ] && cat /tmp/rubocop.log
```

**Comment by dhempy** [comment]
Aside from a unix-y aesthetic of "succeed quietly," I want to alias rspec on my dev box to also run rubocop.  (Because I never remember to run it on my own!)  So I want it to nag me about formatting ASAP, but I don't want to see the clutter when I'm a good boy.

**Comment by dhempy** [comment]
$ rubocop -V
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.5-compliant syntax, but you are running 2.1.4.
0.28.0 (using Parser 2.2.0.2, running on ruby 2.1.4 x86_64-darwin14.0)

**Comment by bbatsov** [comment]
Closing due to lack of activity.

**Comment by pvdb** [comment]
The other use case is when using `rubocop` in `git` pre-commit hooks... without any offenses, you don't really want to see any output, as that's just noise... "succeed quietly" indeed, as @dhempy so correctly pointed out.

**Comment by Drenmi** [comment]
So basically `--format simple`, but you also want to suppress the final:

```
1 file inspected, no offenses detected
```

Correct, @pvdb?

**Comment by pvdb** [comment]
Exactly that, @Drenmi -- "succeed completely quietly", to paraphrase @dhempy :smile:

**Comment by Drenmi** [comment]
Reopening due to renewed interest. Might take a stab at it tonight.

**Comment by dhempy** [comment]
If you want to build on/steal/mutate https://github.com/bbatsov/rubocop/pull/1593 , feel free!

**Comment by pvdb** [comment]
For context, the "succeed quietly" guideline @dhempy was referring to is also known as ["The Rule of Silence"](https://en.wikipedia.org/wiki/Unix_philosophy?section=6#Eric_Raymond.E2.80.99s_17_Unix_Rules), a long-standing, KISS-inspired design rule part of [the Unix philosophy](https://en.wikipedia.org/wiki/Unix_philosophy):

> ** Rule of Silence**
>
> Developers should design programs so that they do not print unnecessary output. This rule aims to allow other programs and developers to pick out the information they need from a program's output without having to parse verbosity.

**Comment by derekcannon** [comment]
I am writing a Rails generator that uses Rubocop to clean up the file output before it generates the files. It would be helpful to have an option to suppress ALL the output.

**Comment by shamess** [comment]
Another use case: I run Rubocop in `guard`, and would like it to not output anything when the successful. No need to push away my failing test output.
